### PR TITLE
Plotting and other updates

### PR DESCRIPTION
**Pull request by gabrieltseng**
Plotting updates from #29 :
- [x] Plot NDVI
- [x] Plot probabilties for the sklearn models
- [x] Ensure all plots use the same colorbar range

In addition:
- [x] I have added 3 runs where we leave out different countries during finetuning, to better understand generalization effects. I also save those models' state dicts
- [x] Fix a bug where we save the non-finetuned model state dict as the finetuned model

| Model | Head | Overall F1 | Latvia F1 |
| --- | ---| --- | --- |
| WorldCereal | CatBoost | 0.8489 | 0.8842 |
| Full | CatBoost | 0.8566 | 0.9051 |
| No Latvia | CatBoost | 0.8558 | 0.8694 |

### CODE REVIEW

File: data/finetuned_model.pt
Comment (kvantricht) [inline data/finetuned_model.pt:1]: @gabrieltseng any idea why the model is 50% smaller?

File: tests/test_dataset.py [hunk 1]
```diff
@@ -96,6 +98,9 @@ def test_combine_predictions(self):
         # Check all predictions between 0 and 1
         self.assertTrue(df_predictions["prediction_0"].min() >= 0)
         self.assertTrue(df_predictions["prediction_0"].max() <= 1)
+        # Check all ndvi values between 0 and 1
```
[the remark this answers is not in the document]
Comment (gabrieltseng) [reply tests/test_dataset.py:101]: Here we are testing the `combine_predictions` function (or in this case the "NDVI array to `.nc`" function).

So there is no normalization. This test is a bit contrived; we are basically just making sure nothing really unexpected happens between the definition of NDVI values (line 79) and its integration into the netcdf file.

[the remark this answers is not in the document]
Comment (kvantricht) [reply tests/test_dataset.py:101]: Oh okay I was overthinking it in this case.

File: data/finetuned_model.pt
Comment (gabrieltseng) [reply data/finetuned_model.pt:1]: Yes - we are now saving the weights for a `PrestoFinetuningModel` (which is an encoder with a classification head on top).

For loading this model:
```python
from presto.presto import Presto
from presto.utils import device
import torch

presto = Presto.load_pretrained()
finetuned_model = presto.construct_finetuning_model(num_outputs=1)
finetuned_model.load_state_dict(torch.load(default_model_path, map_location=device))
```

Comment (gabrieltseng) [reply data/finetuned_model.pt:1]: hmm this should work if we add `strict=False` to the `load_state_dict` function in [`load_pretrained`](https://github.com/WorldCereal/presto-worldcereal/blob/46a768a90a851bcb657a11163613c9dfc1551fdf/presto/presto.py#L795).

I can add that in this PR.

Comment (gabrieltseng) [reply data/finetuned_model.pt:1]: Added in https://github.com/WorldCereal/presto-worldcereal/pull/41/commits/e6d386a9c1ec1d6c570628211bf000d22b0421e8 !
